.github/workflows/ci.yaml

```diff
@@ -8,6 +8,10 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v3
+      # I think setup-python v4 has a builtin cache that might be easier to use!
```

Comment:
Oh cool. If newer versions automatically handle the cache that is much simpler. I will go ahead and update to 3.12 since there's nothing holding us back to an earlier version.